aips/staking-contract-to-delegation-pool.md

```diff
@@ -0,0 +1,99 @@
+---
+aip:
+title: Direct conversion from staking contract to delegation pool
+author: alexfilip2/alexandru@bwarelabs.com
+discussions-to (*optional):
+Status: Draft
+last-call-end-date (*optional):
+type: Standard (Framework)
+created: 02/04/2024
+updated (*optional):
+requires (*optional):
+---
+
+# AIP-X - Direct conversion from staking contract to delegation pool
+
+## Summary
+
+The legacy staking contract cannot be directly converted to a delegation pool. The staker should unlock their entire stake from the staking contract and then delegate it to a newly created delegation pool.
+This results in downtime for the validator, leading to missed rewards and a temporary decrease in network's security.
+Furthermore, the operator needs to setup an extra validator node that must be synced at the time of staker's delegation (in order to achieve the least possible downtime).
+
+The ideal conversion should be atomic: destroy the staking contract and, on top of its underlying stake pool, create a delegation pool of consistent stake ownership and configuration.
```

Comment:
who can call the function to convert to delegation pool? it should be the staker / pool owner of the staking contract

Reply:
Currently, only the staker(owner) of the staking contract can convert it to a delegation pool. However, if the ownership role (who can set operator and commission fee) remains at staker then there is no impact if the operator can call it or the function is public.